chain/chain/src/store.rs

```diff
@@ -1824,26 +1861,41 @@ impl<'a> ChainStoreUpdate<'a> {
             .insert(epoch_hash.clone(), light_client_block);
     }
 
+    // save the outgoing receipts generated by chunk from block `hash` for shard `shard_id`
     pub fn save_outgoing_receipt(
         &mut self,
         hash: &CryptoHash,
         shard_id: ShardId,
-        receipt_result: ReceiptResult,
+        outgoing_receipts: Vec<Receipt>,
     ) {
-        let mut outgoing_receipts = Vec::new();
-        for (receipt_shard_id, receipts) in receipt_result {
-            for receipt in receipts {
-                self.chain_store_cache_update
-                    .receipt_id_to_shard_id
-                    .insert(receipt.receipt_id, receipt_shard_id);
-                outgoing_receipts.push(receipt);
-            }
-        }
         self.chain_store_cache_update
             .outgoing_receipts
             .insert((*hash, shard_id), outgoing_receipts);
     }
```

Comment:
This function does not make sense any more. Why do we only update the cache?

Reply:
The cache is a cache for update, it gets committed to the database when you call ChainStoreUpdate.commit

I do think there are too many layers of caches and updates in ChainStore that it does get confusing